### DISCUSSION TITLE
PIP-22: Agent Node Defination

### DISCUSSION THREAD

**Discussion by ja7ad** [pactus-project/PIPs#94] [PIPs]
PIP-22: [Agent Node Defination](https://pips.pactus.org/PIPs/pip-22)

**Comment by kehiy** [comment]
IMO, we have to keep the agent node more proper. mostly the purpose of this `node` field is to determine which implementation is running there! so, in this case, both GUI and CLI are the same implementations.

so we use the `Pactus` implementation of the Pactus node in our official implementation. but if someone wrote another implementation to a node or a fork of this one, they have to use their name!

for example:

`pacrs` (just a random name)


so, we only have the same name of `pactus` or another name that we can set as a node in the agent. GUI or daemon are the only ways to run official implementation. not different implementation.

> NOTE: we can add the **correct** purpose of `node` field in main PIP and close this PIP since we don't need it.
